src/common.jl

```diff
@@ -209,38 +209,64 @@ compact_repr(num::Number) = repr(num, context = :compact => true)
 
 ceil_neg_log10(x) =
     roundable(-log10(x)) ? ceil(Integer, -log10(x)) : floor(Integer, -log10(x))
-round_up_tick(x, m) = (
-    x == 0 ? 0 :
-    (x > 0 ? ceil(x, digits = ceil_neg_log10(m)) : -floor(-x, digits = ceil_neg_log10(m)))
-)
-round_down_tick(x, m) = (
-    x == 0 ? 0 :
-    (x > 0 ? floor(x, digits = ceil_neg_log10(m)) : -ceil(-x, digits = ceil_neg_log10(m)))
-)
-round_up_subtick(x, m) = (
-    x == 0 ? 0 :
+round_up_tick(x::F, m) where {F} =
     (
-        x > 0 ? ceil(x, digits = ceil_neg_log10(m) + 1) :
-        -floor(-x, digits = ceil_neg_log10(m) + 1)
-    )
-)
-round_down_subtick(x, m) = (
-    x == 0 ? 0 :
+        x == 0 ? 0 :
+        (
+            x > 0 ? ceil(x, digits = ceil_neg_log10(m)) :
+            -floor(-x, digits = ceil_neg_log10(m))
+        )
+    ) |> F
+round_down_tick(x::F, m) where {F} =
     (
-        x > 0 ? floor(x, digits = ceil_neg_log10(m) + 1) :
-        -ceil(-x, digits = ceil_neg_log10(m) + 1)
-    )
-)
-float_round_log10(x::F, m) where {F<:AbstractFloat} = (
-    x == 0 ? F(0) :
+        x == 0 ? 0 :
+        (
+            x > 0 ? floor(x, digits = ceil_neg_log10(m)) :
+            -ceil(-x, digits = ceil_neg_log10(m))
+        )
+    ) |> F
+round_up_subtick(x::F, m) where {F} =
     (
-        x > 0 ? round(x, digits = ceil_neg_log10(m) + 1)::F :
-        -round(-x, digits = ceil_neg_log10(m) + 1)::F
-    )
-)
+        x == 0 ? 0 :
+        (
+            x > 0 ? ceil(x, digits = ceil_neg_log10(m) + 1) :
+            -floor(-x, digits = ceil_neg_log10(m) + 1)
+        )
+    ) |> F
+round_down_subtick(x::F, m) where {F} =
+    (
+        x == 0 ? 0 :
+        (
+            x > 0 ? floor(x, digits = ceil_neg_log10(m) + 1) :
+            -ceil(-x, digits = ceil_neg_log10(m) + 1)
+        )
+    ) |> F
+float_round_log10(x::F, m) where {F<:AbstractFloat} =
+    (
+        x == 0 ? 0 :
+        (
+            x > 0 ? round(x, digits = ceil_neg_log10(m) + 1) :
+            -round(-x, digits = ceil_neg_log10(m) + 1)
+        )
+    ) |> F
 float_round_log10(x::Integer, m) = float_round_log10(float(x), m)
 float_round_log10(x) = x > 0 ? float_round_log10(x, x) : float_round_log10(x, -x)
 
+function unit_str(x, fancy)
+    io = IOContext(PipeBuffer(), :fancy_exponent => fancy)
+    show(io, unit(x))
+    read(io, String)
+end
+number_unit(x::AbstractVector{<:Quantity}, fancy = true) =
+    ustrip.(x), unit_str(first(x), fancy)
+number_unit(x::Quantity, fancy = true) = ustrip(x), unit_str(x, fancy)
+number_unit(x::AbstractVector, args...) = x, nothing
+number_unit(x::Number, args...) = x, nothing
+
+unit_label(label::AbstractString, unit::AbstractString) =
+    label == "" ? unit : "$label ($unit)"
```

Comment:
Can we assume that `label` is already stripped? I can imagine that people would pass something like `label = "      "`

> Should we maybe add parentheses around units even when there is no label

Just my own opinion, I like the current version more.

Reply:
Good point, I've `rstrip`ed the labels before applying units, and added more tests.